modal/client.py

```diff
@@ -136,6 +137,7 @@ async def _close(self, forget_credentials: bool = False):
 
         if forget_credentials:
             self._credentials = None
+            self._snapshotted = True
```

Comment:
Setting `_snapshotted` as a side-effect of the `forget_credentials` parameter seems kinda weird; maybe we could rename that to be more general?

Reply:
I did at one point in this change rename `forget_credentials` to `prepare_for_restore` or something. Is that better?

Reply:
Yeah something like that might make more sense.